test/controllers/declaration/DocumentsProducedControllerSpec.scala

```diff
@@ -57,21 +60,25 @@ class DocumentsProducedControllerSpec
 
     "read item from cache and display it" in {
 
-      val cachedData = DocumentsProducedData(
-        Seq(DocumentsProduced(Some("XA"), Some("XI"), Some("XZ"), Some("YX"), Some("YY"), Some(BigDecimal("9999"))))
-      )
-      withCaching[DocumentsProducedData](Some(cachedData), "DocumentsProducedData")
+      val document = DocumentsProducedSpec.correctDocumentsProduced
+      val cachedData = DocumentsProducedData(Seq(document))
+      withCaching[DocumentsProducedData](Some(cachedData), DocumentsProducedData.formId)
 
       val result = route(app, getRequest(uri)).get
       val page = contentAsString(result)
 
       status(result) must be(OK)
-      page must include("XA")
-      page must include("XI")
-      page must include("XZ")
-      page must include("YX")
-      page must include("YY")
-      page must include("9999")
+      page must include(document.documentTypeCode.get)
```

Comment:
I wouldn't say that repeated assertions make tests unreadable. Yes, they are longer, but you can still figure out quite quickly what is being tested.
The approach you are proposing is shorter but I've tried it and I highly discourage to use it anywhere - the assertion relies on comparing two `String`s. It is extremely time-consuming to find what is wrong if the test fails, because the only information is that `String`s are not equal.

Reply:
And the other thing is, that there is nothing wrong in tests being repetitive if they serve thier purpose.